### PR TITLE
[Gekidou] Channel Info screen

### PR DESCRIPTION
**Pull request by enahum**
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-41575

```release-note
NONE
```

### CODE REVIEW

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! Nothing seemingly blocking, but marking it as request changes because of the amount of comments.

File: app/components/markdown/markdown.tsx [hunk 1]
```diff
@@ -264,11 +282,20 @@ const Markdown = ({
     };
 
     const renderHeading = ({children, level}: {children: ReactElement; level: string}) => {
+        if (disableHeading) {
+            return (
+                <Text style={style.bold}>
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/components/markdown/markdown.tsx:287]: is in the figma design
image

File: app/components/markdown/markdown.tsx [hunk 2]
```diff
@@ -497,7 +533,7 @@ const Markdown = ({
     };
 
     const parser = useRef(new Parser({urlFilter, minimumHashtagLength})).current;
-    const renderer = useRef(createRenderer()).current;
+    const renderer = useMemo(createRenderer, [theme, textStyles]);
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/components/markdown/markdown.tsx:536]: block styles are based on the theme, imageMetadata will not change, nor any of the others.. in fact textStyles if changed is also cause of a theme change

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
Awesome work @enahum. It's looking great. Here are a few things I found:

1. I got in to a scenario where I double tapped the channel name to open the channel modal. Two copies of the modal opened up. I closed one, but then the last one couldn't be closed.
2. If I trigger the Channel info modal from the channel quick actions bottom sheet, there is no `X` to close the modal (although I can dismiss by swiping down). If I open the channel info model by tapping the channel name, the `X` _does_ show. We should make sure the `X` is there in all cases.
3. The scroll bar seems to interfere with the content. Can we make sure it stays anchored closer to the right edge? (see screenshot of what I mean)



4. The label for the `Leave channel` menu item seems to be misaligned with the other menu item label


5. When I go to the 'Edit channel' view, all the inputs are in the focused state for all inputs that have values filled in. Should we file this separately?


6. For DM info modal, the font size for the user's title is too large. Can we make sure it is 16?


7. For DM and GM info modal, the `Close direct message` menu option is misaligned
8. It looks like there is more space above and below the divider lines than in the design. Can we reduce the space by 12px above and below? But keep the spacing as-is in the top section between the 'Created by' line and the first divider line.
9. On tablet, I think we need a bit more margin at the bottom of the modal to ensure the last few items aren't so tight to the bottom. Can we add 16px margin at the bottom?
10. On tablet, the 'Copy link' button isn't spanning the same width as the other buttons.
IMG_D369A71ADB1B-1

**Comment by enahum** [comment]
@matthewbirtch all items should be correct now

**Comment by matthewbirtch** [comment]
Still seeing these ones:

>  The label for the `Leave channel` menu item seems to be misaligned with the other menu item label

>  For DM and GM info modal, the `Close direct message` menu option is misaligned

Everything else looks great to me

**Comment by enahum** [comment]
> Still seeing these ones:
> 
> > The label for the `Leave channel` menu item seems to be misaligned with the other menu item label
> 
> > For DM and GM info modal, the `Close direct message` menu option is misaligned
> 
> Everything else looks great to me

done

**Review by matthewbirtch — APPROVED** [review]

**Pull request merged**